pgjdbc/src/test/java/org/postgresql/test/core/v2/V2ParameterListTests.java

```diff
@@ -0,0 +1,94 @@
+/*-------------------------------------------------------------------------
+ *
+ * Copyright (c) 2003-2016, PostgreSQL Global Development Group
+ *
+ *
+ *-------------------------------------------------------------------------
+ */
+
+package org.postgresql.test.core.v2;
+
+import org.postgresql.core.ParameterList;
+
+import junit.framework.TestCase;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+
+/**
+ * Test cases to make sure the parameterlist implementation works as expected.
+ * Test case located in different package as the package private implementation.
+ * Reflection used to crack open and access the class. Not pretty but works.
+ *
+ * @author Jeremy Whiting jwhiting@redhat.com
+ *
+ */
+public class V2ParameterListTests extends TestCase {
+
+  public void testMergeOfParameterLists() {
+    try {
+      ClassLoader cl = this.getClass().getClassLoader();
+      Class cls = Class.forName("org.postgresql.core.v2.SimpleParameterList",
+          true, cl);
+      Constructor c = cls.getDeclaredConstructor(Integer.TYPE, Boolean.TYPE);
+      c.setAccessible(true);
+      Object o1SPL = c
+          .newInstance(new Object[] { new Integer(8), Boolean.TRUE });
+      assertNotNull(o1SPL);
+      Method msetIP = cls.getMethod("setIntParameter", Integer.TYPE,
+          Integer.TYPE);
+      assertNotNull(msetIP);
+      msetIP.setAccessible(true);
```

Comment:
In most situations I agree using reflection in test cases shouldn't be necessary.
 However, in this case the test classes package all start with the "test." making the package private source code group inaccessible. Both org.postgresql.core.v2.SimpleParameterList and org.postgresql.core.v3.SimpleParameterList were set package private by another developer.
 I didn't want to go through the existing codebase making more public accessibility changes than necessary. Using reflection preserved the original intent where possible.